clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1023,6 +1023,28 @@ public void onAssignment(Set<TopicPartition> assignment) {
         sensors.updatePartitionLagAndLeadSensors(assignment);
     }
 
+    /**
+     * Clear the buffered data (completed fetches) for all topics
+     */
+    public void clearBufferedData() {
+        completedFetches.clear();
+    }
+
+    /**
+     * Clear the buffered data (completed fetches) for a set of {@link TopicPartition}
+     */
+    public void clearBufferedDataForTopicPartitions(Set<TopicPartition> topicPartitions) {
```

Comment:
nit: naming topicPartitions as topicPartitionsToClear would make the code more readable

Reply:
done